app/messaging/schemas/remaining-amount.js

```diff
@@ -0,0 +1,8 @@
+const Joi = require('joi')
+
+module.exports = Joi.number().integer().required()
+  .messages({

```

Comment:
Missing `any.required`

Reply:
remainingAmount is not anything but number

Reply:
why do we need to add any?

Reply:
It's the error type `Joi` gives when you have `any` i.e. `number.required()` , `string.required()`, etc and you don't incldue it: https://joi.dev/api/?v=17.9.1#anyrequired---aliases-exist

Reply:
So in our case we do not want any (general) but number(specific).

Reply:
Yes, but tht's not how Joi error types works

Reply:
How does Joi error work? We already know the specific here, we are not testing for multiple sub-set.

Reply:
So whenever we create this Joi schema, we are building a set of validation rules.

When Joi attempts to check the inputted value against the validation rules, it does so in some order (not too important). If it fails for a rule, it creates an error key in the returned object w/ lots of information including  a 'type', i.e. which rule part was invalidated and a message which is what we're overriding here.

Reply:
Could you please provide an example where general check is important when we already know the specific?

Reply:
It's how Joi works, you don't get a `number.required` or `string.required` type its `any.required`.

https://joi.dev/api/?v=17.9.1#anyrequired